### PR TITLE
src: remove unused headers-exported functions

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None : generic cleaning

Describe changes:
- remove functions defined in a header and never used (debug, api-completeness, and others)
- remove double definition of IPPairLock

Python helper script to do this :
```
import subprocess
import sys
import os

def patchFuncStatic(funcname, cfile, hfile):
    subprocess.run(["sed", "-i", "-e", "/%s/d" % funcname, hfile], stdout=subprocess.PIPE, text=True)
    f = open(cfile, "r")
    f2 = open(cfile+".new", "w")
    writing = True
    for l in f.readlines():
        if funcname in l:
            writing = False
            continue
        if writing:
            f2.write(l)
        elif l.rstrip() == '}':
            writing = True
    f.close()
    f2.close()
    os.rename(cfile+".new", cfile)

def testFuncName(funcret, funcname, hfile):
    result = subprocess.run(["git", "grep", funcname], stdout=subprocess.PIPE, text=True)
    d = dict()
    nbh = 0
    for l in result.stdout.split('\n'):
        hasf = l.split(':')[0]
        if len(hasf) == 0:
            continue
        if hasf == hfile:
            nbh = nbh + 1
            continue
        d[hasf] = d.get(hasf, 0)+1
    if nbh == 1 and len(d) == 1 and list(d.values())[0] == 1:
        print(funcret, funcname, d)
        patchFuncStatic(funcname, list(d.keys())[0], hfile)

f = open(sys.argv[1], "r")
for l in f.readlines():
    if '(' in l:
        ps = l.split('(')[0].split()
        if len(ps) == 2:
            funcret = ps[0]
            funcname = ps[1]
            if funcret != '/*' and funcret != '#define' and funcret != '*':
                if funcname[0] == '*':
                    funcname=funcname[1:]
                testFuncName(funcret,funcname,sys.argv[1])
```

Needed rebase of #8887

### CODE REVIEW

File: src/detect-engine.c [hunk 1]
```diff
@@ -186,53 +186,6 @@ void DetectPktInspectEngineRegister(const char *name,
     }
 }
 
-/** \brief register inspect engine at start up time
- *
- *  \note errors are fatal */
-void DetectFrameInspectEngineRegister(const char *name, int dir,
```
Comment (catenacyber) [inline src/detect-engine.c:192]: Is it normal that this is unused ?

File: src/ippair.c [hunk 2]
```diff
@@ -524,11 +519,6 @@ void IPPairRelease(IPPair *h)
     SCMutexUnlock(&h->m);
 }
 
-void IPPairLock(IPPair *h)
-{
-    SCMutexLock(&h->m);
-}
```
Comment (catenacyber) [inline src/ippair.c:530]: Looks strange that `IPPairLock` is unused but `IPPairUnlock` is used

File: src/util-storage.c [hunk 3]
```diff
@@ -356,197 +356,3 @@ void StorageFree(Storage **storage, StorageEnum type)
...
-    }
-
-    StorageCleanup();
-    return 1;
-error:
-    StorageCleanup();
-    return 0;
-}
-
-void StorageRegisterTests(void)
```
Comment (catenacyber) [inline src/util-storage.c:546]: Should these be run ?

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on build_asan.
  

Pipeline 14379

**Comment by catenacyber** [comment]
Rebased in https://github.com/OISF/suricata/pull/9099

**Pull request closed without merging**